hardware/esp8266com/esp8266/libraries/ESP8266WiFi/examples/ConnectWithWPA/ConnectWithWPA.ino

```diff
@@ -0,0 +1,79 @@
+/*
+ * This example connects to a WPA encrypted Wifi network.
+ * Then it prints the  MAC address of the ESP8266,
+ * the IP address obtained, and other network details.
+ *
+ * created 13 July 2010
+ * by dlf (Metodo2 srl)
+ * modified 5 July 2015
+ * by Mohamed Elhariry
+ */
+
+#include <ESP8266WiFi.h>
+
+char ssid[] = "yourNetwork";
+char password[] = "secretPassword";
+int status = WL_IDLE_STATUS;     // the Wifi radio's status
+
+void setup() {
+  //Initialize serial and wait for port to open:
+  Serial.begin(115200);
+
+  // attempt to connect to Wifi network:
+  while (true) {
+    Serial.print("Attempting to connect to WPA SSID: ");
+    Serial.println(ssid);
+
+    // Connect to WEP/WPA/WPA2 network:
+    WiFi.begin(ssid, password);
```

Comment:
Please bear in mind if you use WiFi.begin() alone, it will try any network you previously connected to
in other runs.

That's useful as a sketch because people will get connected to their know wifi first.